### ISSUE TITLE
Malformed Credentials when using maven.repository with user and password

### ISSUE THREAD

**Issue by davidmandle** [bazelbuild/rules_jvm_external#720]
When using `maven.repository` to define a private repository, e.g. with

```
    repositories = [
        maven.repository(
            "https://some.private.maven.re/po",
            password = "example-password",
            user = "johndoe",
        ),
    ],
```

Build fails with "Error while fetching artifact with coursier: Malformed credentials"

Inspecting java_argsfile shows that it has these lines appended to it:

```
--credentials
some.private.maven.re johndoe:example-password
```

I believe the failure is because Coursier expects "some.private.maven.re johndoe:example-password" as one argument, but the argsfile is passing it as two separate arguments.

**Comment by davidmandle** [comment]
The root cause seems to be https://github.com/bazelbuild/rules_jvm_external/pull/589, the change to use [Command-Line Argument Files](https://docs.oracle.com/en/java/javase/15/docs/specs/man/java.html#java-command-line-argument-files) for Java 9 and higher. This change creates an argument file, writing each argument to a different line.  But the Command-Line Argument Files documentation states:

> Any option that contains spaces, such as a path component, must be within quotation marks using quotation ('"') characters in its entirety.

This means that arguments containing whitespace (including the --credentials argument) are being passed as separate arguments.

One possible solution is to wrap all arguments in quotes (escaping any quotes inside) before writing them to the arguments file.

**Comment by shs96c** [comment]
Closed by #721
